nncf/torch/composite_compression.py

```diff
@@ -132,3 +133,27 @@ def compression_rate(self) -> float:
                 sum_compression_rate += sum_compression_rate
                 not_none_compression_rate_cnt += 1
         return sum_compression_rate / max(not_none_compression_rate_cnt, 1)
+
+    def prepare_for_inference(self, save_original_model: bool = True) -> NNCFNetwork:
```

Comment:
@AlexKoff88 @vshampor What you think about this implementation of api?
`prepare_of_inference.py` removed. Logic moved to algo controller. 
 - https://github.com/openvinotoolkit/nncf/pull/1526/files#diff-98e7d8799a3972a5bc5e857caad2fa52f69f4282d80ac3925d67faa25c11de39R661
 - https://github.com/openvinotoolkit/nncf/pull/1526/files#diff-a5e2ce6b7b41ac95fd0e90302455ad07cef0076e3b2eeba8c4d11319a6b80c8fR1453



But need to use protected-access to "model" property to save instance of the model. 
https://github.com/openvinotoolkit/nncf/pull/1526/files#diff-835a066a208019a262cb822315aa79cb087e6a2d4974a7e8c37b0b4f3e3a043dR153-R155

Reply:
Does it mean that each algorithm makes a copy of the model that it has a reference to and applies modifications fo this copy? I would expect that copy of the model is made once and each algo does corresponding modifications. @AlexanderDokuchaev, @vshampor, what do you think? is it technically possible within the current architecture?

Reply:
I am ok with this API.

Reply:
> Does it mean that each algorithm makes a copy of the model that it has a reference to and applies modifications fo this copy?

No, makes copy only once. But to avoid multiple copies of the model, I have to add to a modified protected property in `PTCompositeCompressionAlgorithmController`.


https://github.com/openvinotoolkit/nncf/blob/38a123a3d1fc2867918e4f1ce141e7f15d5092c7/nncf/torch/composite_compression.py#L153-L155